crates/rspack_core/src/options/module.rs

```diff
@@ -636,6 +637,53 @@ impl RuleSetCondition {
       Self::Func(f) => f(data).await,
     }
   }
+
+  #[async_recursion]
+  async fn match_when_empty(&self) -> Result<bool> {
+    let res = match self {
+      RuleSetCondition::String(s) => s.is_empty(),
+      RuleSetCondition::Regexp(rspack_regex) => rspack_regex.test(""),
+      RuleSetCondition::Logical(logical) => logical.match_when_empty().await?,
+      RuleSetCondition::Array(arr) => {
+        arr.is_empty() && try_any(arr, |c| async move { c.match_when_empty().await }).await?
+      }
+      RuleSetCondition::Func(func) => func("".into()).await?,
+    };
+    Ok(res)
+  }
+}
+
+#[derive(Debug)]
+pub struct RuleSetConditionMatch {
+  condition: RuleSetCondition,
+  match_when_empty: RwLock<Option<bool>>,
```

Comment:
```suggestion
  match_when_empty: OnceLock<bool>,
```

Seems `OnceLock` is more suitable for this case